src/Avalonia.Visuals/Rendering/SceneGraph/BrushDrawOperation.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Copyright (c) The Avalonia Project. All rights reserved.
+// Licensed under the MIT license. See licence.md file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using Avalonia.Media;
+using Avalonia.Platform;
+using Avalonia.VisualTree;
+
+namespace Avalonia.Rendering.SceneGraph
+{
+    /// <summary>
+    /// Base class for draw operations that can use a brush.
+    /// </summary>
+    internal abstract class BrushDrawOperation : IDrawOperation
+    {
+        /// <inheritdoc/>
+        public abstract Rect Bounds { get; }
+
+        /// <inheritdoc/>
+        public abstract bool HitTest(Point p);
+
+        /// <summary>
+        /// Gets a collection of child scenes that are needed to draw visual brushes.
+        /// </summary>
+        public abstract IDictionary<IVisual, Scene> ChildScenes { get; }
+
+        /// <inheritdoc/>
+        public abstract void Render(IDrawingContextImpl context);
+
+        /// <summary>
+        /// Converts a possibly mutable brush to an immutable brush.
+        /// </summary>
+        /// <param name="brush">The brush.</param>
+        /// <returns>An immutable brush</returns>
+        protected IBrush ToImmutable(IBrush brush)
```

Comment:
The `ToImmutable` method that it calls isn't an extension method - it's defined by `IMutableBrush` and is a method on the brush itself.